flow/layers/layer.h

```diff
@@ -297,29 +300,49 @@ class Layer {
     // By default, the layer is saved on and restored from
     // `internal_nodes_canvas`. The `save_mode` parameter can be modified to
     // save the layer on other canvases.
-    [[nodiscard]] static AutoSaveLayer Create(
+    [[nodiscard]] static SkAutoSaveLayer Create(
         const PaintContext& paint_context,
         const SkCanvas::SaveLayerRec& layer_rec,
         SaveMode save_mode = SaveMode::kInternalNodesCanvas);
 
-    ~AutoSaveLayer();
-
    private:
-    AutoSaveLayer(const PaintContext& paint_context,
-                  const SkRect& bounds,
-                  const SkPaint* paint,
-                  SaveMode save_mode = SaveMode::kInternalNodesCanvas);
+    SkAutoSaveLayer(const PaintContext& paint_context,
+                    const SkRect& bounds,
+                    const SkPaint* paint,
+                    SaveMode save_mode = SaveMode::kInternalNodesCanvas);
 
-    AutoSaveLayer(const PaintContext& paint_context,
-                  const SkCanvas::SaveLayerRec& layer_rec,
-                  SaveMode save_mode = SaveMode::kInternalNodesCanvas);
+    SkAutoSaveLayer(const PaintContext& paint_context,
+                    const SkCanvas::SaveLayerRec& layer_rec,
+                    SaveMode save_mode = SaveMode::kInternalNodesCanvas);
 
-    const PaintContext& paint_context_;
     const SkRect bounds_;
+
+    const PaintContext& paint_context_;
```

Comment:
Why did you move this field? It just led to a bunch of unnecessary diff lines in a number of places.